config/management/src/validator_config.rs

```diff
@@ -38,10 +40,13 @@ impl ValidatorConfig {
         validator_address: NetworkAddress,
         reconfigure: bool,
     ) -> Result<Transaction, Error> {
-        let mut storage = StorageWrapper::new(
-            self.validator_backend.name(),
-            &self.validator_backend.validator_backend,
-        )?;
+        let config = self
+            .config
+            .load()?
+            .override_chain_id(self.chain_id)
+            .override_validator_backend(&self.validator_backend.validator_backend)?;
```

Comment:
Should we do some smart checks and say if you have the ChainId with the config path to reject it?

Reply:
I had thought about it, but I'm kinda okay with overrides. If folks wanna do weird stuff, it makes tool writing really a pain.